### ISSUE TITLE
CAD: Track stuck or not moving

### ISSUE THREAD

**Issue by fzeiser** [fzeiser/OCL#19]
Implemented the geometry with CAD files between commits 638ba43a526e34c80f51c800d98a642e964e624b and 82698892ae0737b1faaaedd69d808c0c255a4b85.

When running the simulation one gets following type of warnings issued:
```
An example of a warning message:
-------- WWWW ------- G4Exception-START -------- WWWW -------
*** G4Exception : GeomNav1002
      issued by : G4Navigator::ComputeStep()
Track stuck or not moving.
          Track stuck, not moving for 10 steps
          in volume -FrameBaseCADphys- at point (-435.204,-1200,-387.442) (local point (-387.442,0,365.204))
          direction: (-0.245012,-0.906444,-0.343989) (local direction: (-0.343989,-0.906444,0.245012)).
          Potential geometry or navigation problem !
          Trying pushing it of 1e-07 mm ...Potential overlap in geometry!

*** This is just a warning message. ***
-------- WWWW -------- G4Exception-END --------- WWWW -------
```

Comparing the output of the simulations to the experiment, I needed to scale it up almost 15 times as much as I did for the simulations without CAD geometry. So it seems like particles really get stuck somewhere. Also, the results look quite weird (much less agreement to the experiment then before):
![c1](https://user-images.githubusercontent.com/15156776/37515401-f86aebca-290a-11e8-8d3e-a01d04219751.png)

I've tried to trace back the error, but was not successful.

Running through a geometry check with `/geometry/test/run` yields quite unsuspicious output:
/geometry/test/run
Running geometry overlaps check...
```
Checking overlaps for volume LaBr3Housing ... OK!
Checking overlaps for volume OCLDetector ... OK!
Checking overlaps for volume Coating ... OK!
Checking overlaps for volume CoatingPlexiGlas ... OK!
Checking overlaps for volume Reflector ... OK!
Checking overlaps for volume Crystal ... OK!
Checking overlaps for volume PlexiGlasWindow ... OK!
Checking overlaps for volume PMTWindow ... OK!
Checking overlaps for volume Cathode ... OK!
[...]
Checking overlaps for volume FrameBallCADphys ... OK!
Checking overlaps for volume FrameRingCADphys ... OK!
Checking overlaps for volume FrameBaseCADphys ... OK!
Checking overlaps for volume ChamberTube ... OK!
Checking overlaps for volume LadderHolder ... OK!
Checking overlaps for volume Target ... OK!
Checking overlaps for volume TargetHolder ... OK!
Checking overlaps for volume TargetHole ... OK!
Checking overlaps for volume SiRiHolder ... OK!
Checking overlaps for volume SiRi_pad ... OK!
Checking overlaps for volume SiRi_front ... OK!
Checking overlaps for volume dE_strip_cp ... OK!
Checking overlaps for volume dE_strip_cp ... OK!
Checking overlaps for volume dE_strip_cp ... OK!
Checking overlaps for volume dE_strip_cp ... OK!
Checking overlaps for volume dE_strip_cp ... OK!
Checking overlaps for volume dE_strip_cp ... OK!
Checking overlaps for volume dE_strip_cp ... OK!
Checking overlaps for volume dE_strip_cp ... OK!
Checking overlaps for volume SiRi_back ... OK!
Checking overlaps for volume SiRiCable ... OK!
Checking overlaps for volume SiRiCableCu ... OK!
[...]
Geometry overlaps check completed !
```

The geometry output of the latest commit 82698892ae0737b1faaaedd69d808c0c255a4b85. looks like this: 
![cadwhole](https://user-images.githubusercontent.com/15156776/37515045-e29a2e92-2909-11e8-9fa6-37aa48351eef.png)

**Comment by fzeiser** [comment]
The CAD files were converted to STL files with Autodesk Inventor. One possible problem could be following indication when opening the mesh in **freecad**:

> The mesh data structure has some defects

This warning message does not appear, eg, if I open the CAD files of AFRODITE.

I tried to repair the STL files, eg using the inbuild option
Mesh Design> Meshes >  Analyze > Evaluate & Repair
![repair_mesh](https://user-images.githubusercontent.com/15156776/37761990-7c8329ea-2dbb-11e8-8646-00a81f73f9da.png)

However, if one just repairs everything, this leads to wrong orientation of normal vectors. In the picture below one can see that some faces that should be outwards turned, are now turned inwards (black)
![repair_mesh2](https://user-images.githubusercontent.com/15156776/37761989-7c590962-2dbb-11e8-930b-147ff4487975.png)

Even if I omitted the correction of non-manifolds, the problems with Geant4 remained.

**Comment by fzeiser** [comment]
I also tried to fix the problem with **meshlab**.
When opening Meshlab, I'm immediately asked whether I should "`Unify Duplicated Vertices`". I tried this, and also `Merge Close faces`, `remove dublicate vertices`, and `remove dublicate faces` from
Filters > Cleaning and Repairing

Didn't help either.

**Comment by fzeiser** [comment]
Fixed through #22
